cmd/edgectl/aes_install.go

```diff
@@ -0,0 +1,290 @@
+	for {
+		var err error
+		ipAddress, err = i.CaptureKubectl("get IP address", "get", "-n", "ambassador", "service", "ambassador", "-o", `go-template={{range .status.loadBalancer.ingress}}{{print .ip "\n"}}{{end}}`)
+		if err != nil {
+			return err
+		}
+		ipAddress = strings.TrimSpace(ipAddress)
+		if ipAddress != "" {
+			break
+		}
+		time.Sleep(250 * time.Millisecond) // FIXME: Time out at some point...
+	}
+
+	fmt.Println("\nYour IP address is", ipAddress)
+
+	// Wait for Ambassador to be ready to serve ACME requests.
+	// FIXME: This assumes we can connect to the load balancer. If this
+	// assumption is incorrect, this code will loop forever.
+	for {
+		// FIXME: Time out at some point...
+		time.Sleep(500 * time.Millisecond)
+		resp, err := http.Get("http://" + ipAddress + "/.well-known/acme-challenge/")
+		if err != nil {
+			fmt.Printf("Waiting for Ambassador (get): %#v\n", err)
+			continue
+		}
+		_, _ = ioutil.ReadAll(resp.Body)
+		resp.Body.Close()
+		if resp.StatusCode != 404 {
+			fmt.Printf("Waiting for Ambassador: wrong status code: %d\n", resp.StatusCode)
+			continue
+		}
+		if resp.Header.Get("server") != "envoy" {
+			fmt.Printf("Waiting for Ambassador: wrong server header: %s\n", resp.Header.Get("server"))
+			continue
+		}
+		break
+	}
+
+	// Send a request to acquire a DNS name for this cluster's IP address
+	regURL := "https://metriton.datawire.io/beta/register-domain"
+	emailAddress := "ark3+eci@datawire.io"
+	buf := new(bytes.Buffer)
+	_ = json.NewEncoder(buf).Encode(registration{emailAddress, ipAddress})
+	resp, err := http.Post(regURL, "application/json", buf)
+	if err != nil {
+		return errors.Wrap(err, "acquire DNS name (post)")
+	}
+	content, err := ioutil.ReadAll(resp.Body)
+	resp.Body.Close()
+	if err != nil {
+		return errors.Wrap(err, "acquire DNS name (read body)")
+	}
+
+	if resp.StatusCode == 200 {
+		hostname := string(content)
+		fmt.Println("-> Acquiring DNS name", hostname)
+
+		// Wait for DNS to propagate. This tries to avoid waiting for a ten
+		// minute error backoff if the ACME registration races ahead of the DNS
+		// name appearing for LetsEncrypt.
+		for {
+			conn, err := net.Dial("tcp", hostname+":443")
+			if err == nil {
+				conn.Close()
+				break
+			}
+			// fmt.Printf("Waiting for DNS: %#v\n", err)
+			time.Sleep(500 * time.Millisecond)
+		}
+		fmt.Println("-> Automatically configuring TLS")
+		fmt.Println("Please enter an email address. We'll use this email address to notify you prior to domain and certification expiration [None]:", emailAddress)
+		fmt.Println("FIXME: let the user enter an address")
+		// Create a Host resource
+		hostResource := fmt.Sprintf(hostManifest, hostname, namespace, hostname, emailAddress)
+		kargs, err := i.kubeinfo.GetKubectlArray("apply", "-f", "-")
+		if err != nil {
+			return errors.Wrapf(err, "cluster access for install AES")
+		}
+		fmt.Println("\n$ kubectl apply -f - < [Host Resource]")
+		cmd := exec.Command("kubectl", kargs...)
```

Comment:
Should we use a client wrapper here, in case `kubectl` is not in the user's path?

Reply:
Oh definitely. This and `edgectl license` should both avoid using `kubectl` if possible. But that's far lower priority (in my opinion) than getting all the story features and required robustness in place. Do you disagree? Using `kubectl` in the short run can save us a lot of time figuring out the details of how to do the same things it does for us.

Reply:
If `edgectl license` is already dependent on `kubectl`, then let's keep on iterating fast here and revisit this later.